caproto/_hub.py

```diff
@@ -586,6 +586,15 @@ def _fill_defaults(self, data_type, data_count):
             data_count = self.native_data_count
         return data_type, data_count
 
+    def state_changed(self, new_state):
```

Comment:
It might be useful to have the old state too. I usually see this done as `self, new_state, old_state`.